### PR TITLE
Cancel Pending Request

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
Allows cloud to cancel long running requests in case nobody is waiting for them meanwhile. Long overdue.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Review by stelfrag — APPROVED** [review]

**Comment by MrZammler** [comment]
Also should we add it to `msg_name_to_protomsg` so we can have it in `aclk.log` ?

**Review by thiagoftsm — APPROVED** [review]
@underhood , please, take a look in @MrZammler comments. I am approving the PR, because tests worked as expected, LGTM!

**Comment by stelfrag** [comment]
We need a new capability for the cancel request

File: aclk/aclk_rx_msgs.c [hunk 1]
```diff
@@ -446,6 +447,23 @@ int stop_streaming_contexts(const char *msg, size_t msg_len)
...
+    struct aclk_cancel_pending_req cmd;
+    if(parse_cancel_pending_req(msg, msg_len, &cmd)) {
+        error_report("Error parsing CancelPendingReq");
+        return 1;
+    }
+
+    log_access("ACLK CancelPendingRequest REQ: %s, cloud trace-id: %s", cmd.request_id, cmd.trace_id);
+
+    if (mark_pending_req_cancelled(cmd.request_id))
+        error_report("CancelPending Request for %s failed. No such pending request.", cmd.request_id);
```
[the remark this answers is not in the document]
Comment (underhood) [reply aclk/aclk_rx_msgs.c:461]: updated

**Comment by underhood** [comment]
resolved comments from @MrZammler and @stelfrag

**Review by stelfrag — APPROVED** [review]

**Review by MrZammler — APPROVED** [review]

**Pull request merged**